sources/csv_reader/__init__.py

```diff
@@ -0,0 +1,64 @@
+"""
+Source, which uses the `filesystem` source and DuckDB
+to extract CSV files data from the given locations.
+"""
+from typing import Iterable, List
+
+import duckdb
+import pendulum
+
+import dlt
+from dlt.common.typing import TDataItem, TAnyDateTime
+from filesystem import filesystem
+from fsspec.implementations.local import LocalFileSystem
+
+from .helpers import add_columns
+
+
+@dlt.resource
+def read_location(files):
+    """A resource to extract data from the given CSV files.
+
+    Args:
+        files (List[FileItem]): A list of files to read.
+
+    Returns:
+        Iterable[TDataItem]: Data items, read from the given CSV files.
+    """
+    state = dlt.current.resource_state()
+    start_from = state.setdefault("last_modified", pendulum.datetime(1970, 1, 1))
+
+    results = []
+    for file in files:
+        if file["modification_date"] <= start_from:
+            continue
```

Comment:
Incremental
Files are not sorted by dates, and some storages doesn't even allow that, so we can't use the built-in incremental